frontend/app/components/progress.tsx

```diff
@@ -26,13 +26,17 @@ const indicatorBaseClassName = 'h-full w-full flex-1 transition-all';
 export interface ProgressProps extends React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> {
   size?: keyof typeof sizes;
   variant?: keyof typeof variants;
+  label?: string;
 }
 
 const Progress = React.forwardRef<React.ElementRef<typeof ProgressPrimitive.Root>, ProgressProps>(({ className, size = 'base', variant = 'default', value, ...props }, ref) => {
   return (
-    <ProgressPrimitive.Root ref={ref} className={cn(rootBaseClassName, sizes[size], className)} data-testid="progress-root" value={value} {...props}>
-      <ProgressPrimitive.Indicator className={cn(indicatorBaseClassName, variants[variant])} style={{ transform: `translateX(-${100 - (value ?? 0)}%)` }} data-testid="progress-indicator" />
-    </ProgressPrimitive.Root>
+    <>
+      {props.label && <p id="progress-label">{`${props.label} ${value}%`}</p>}
```

Comment:
You need to use `i18next` for the percentage because French has a space between the value and `%` sign. We would use a  non-breaking space and not a space of course.

```js
const formatPercentage = (value, locale) => {
    return Intl.NumberFormat(locale, {
        style: "percent"
    }).format(value);
};

console.log(formatPercentage(1, 'en-CA')); 	// => "100%"
console.log(formatPercentage(0.1, 'en-CA')); 	// => "10%"
console.log(formatPercentage(1, 'fr-CA')); 	// => "100 %"
console.log(formatPercentage(0.1, 'fr-CA')); 	// => "10 %"
```